### ISSUE TITLE
If i attack a chicken, it takes no action, why?

### ISSUE THREAD

**Issue by firnsan** [cuberite/cuberite#3318]
Client version:   1.9.4
Server OS:         Linux
Commit id:        6a6f0fb9
### Expected behavior

when i am playing in singleplayer mode, if i attack a chicken, it would run away from me quickly, flying (or jumping). 
### Actual behavior

But when i am playing with cuberite server,  if i attack a chicken, it  takes no action, specially when i leading it with seeds on my hands, it just follows me for the seeds.
### Steps to reproduce the behavior

None.

**Comment by NiLSPACE** [comment]
This is because it's not implemented yet. It should be allot easier to implement once #3295 is merged though.

**Comment by firnsan** [comment]
That's great. It's worthy to look forward.
By the way, why not anti-compile the vanilla server, and translate to cpp?

**Comment by Pokechu22** [comment]
> By the way, why not anti-compile the vanilla server, and translate to cpp? 

While that would be possible (or rather, it could be decompiled and then compiled natively as java with GCC), there are some legal issues with redistributing mojang's code and also, the point of the project is to create a custom implementation of the Minecraft server (including new features such as the lua API), not just rewrite the original server in C++.

**Comment by NiLSPACE** [comment]
Not to mention that the architecture for Cuberite and vanilla are incompatible.

**Comment by firnsan** [comment]
I appreciate this project very much, specially it's  high performance.  
But when i try to run it in background, it seems to crash, how should i do?
```
nohup ./Cuberite_debug & 
```

the nohup.out's  content:
```
Log: [319acecd2bcfa5ef|14:22:33] Allowed mob: zombiepigman
Log: [319acecd2bcfa5ef|14:22:34] Starting deadlock detector...
Log: [319acecd2bcfa5ef|14:22:34] Finalising startup...
Log: [319acecd2bcfa5ef|14:22:34] Failed to create IPv6 MainSock: 97 (Address family not supported by pr
otocol)
Log: [319acecd2bcfa5ef|14:22:34] Starting WebAdmin...
Log: [319acecd2bcfa5ef|14:22:34] Failed to create IPv6 MainSock: 97 (Address family not supported by pr
otocol)
Log: [319acecd2bcfa5ef|14:22:34] Starting InputThread...
Log: [319acecd2bcfa5ef|14:22:34] Startup complete, took 1785ms!
Log: [319acecd2bcfa5ef|14:22:34] Stopping WebAdmin...
Log: [319acecd2bcfa5ef|14:22:34] Shutting down server...
Log: [319acecd2bcfa5ef|14:22:34] Waiting for thread cMojangAPI::cUpdateThread to finish
Log: [319acecd2bcfa5ef|14:22:34] Thread cMojangAPI::cUpdateThread finished
...
...
Log: [319acecd2bcfa5ef|14:22:34] Destroying plugin "Core".
Log: [319acecd2bcfa5ef|14:22:34] Destroying plugin "ChatLog".
Log: [319acecd2bcfa5ef|14:22:34] Cleaning up...
Log: [319acecd2bcfa5ef|14:22:34] Waiting for thread ServerTickThread to finish
Warning: [319acecd2bcfa5ef|14:22:34] Couldn't notify the input thread; the server will hang before shut
down!
Log: [319acecd2bcfa5ef|14:22:34] Shutdown successful!
Log: [319acecd2bcfa5ef|14:22:34] --- Stopped Log ---
Log: [319acecd2bcfa5ef|14:22:34] Waiting for thread DeadlockDetect to finish
Log: [319acecd2bcfa5ef|14:22:34] Thread DeadlockDetect finished
```

**Comment by Cl1608Ho** [comment]
I am ~~not~~ pretty sure, cuberite has a problem dealing with no stdin. 

From the manage of nohup (on Ubuntu 15.10): 

> If standard input is a terminal, redirect it from /dev/null

Appearantly cuberite sees that the stdin is closed (because the whole purpose of /dev/null is (if you read) emitting instantly EOF (end of file)) and thinks it should stop (same if you run it interactivly and do CTRL+d) 

I have a fix, however I am not 100% sure if it doesn't have any side effects. 
Run your command with `nohup ./Cuberite_debug </dev/zero &`. Now, cuberite will not get /dev/null as stdin but /dev/zero. The difference between them is that /dev/null outputs instantly EOF while /dev/zero outputs \0 (zero characters). I am not sure if this impacts system performance or does anything (because the kernel may be tempted to generate a lot of \0 characters and send them to Cuberite) but in a quick test it *worked*

Maybe @core-team a config value would be nice to read from a fifo instead of stdin. 
Alternatively, what would be even cooler one or two fifo as a local 'api' for running commands and getting the output of just this command ;)

**Comment by madmaxoft** [comment]
I can't remember the reason for specifically quitting when the stdin reports EOF, but there was one.

`/dev/zero` might not be a good idea, because Cuberite tries to read from stdin until it finds a newline; this might fill in the RAM by storing the input string of zeroes. Is there instead a stream that blocks on any read attempt (i. e. says "yes, I will read it in a moment" forever)? That would be the best alternative.

Reading from other pipes is not needed in my opinion, you can already do that with stdin redirection. However, it might be useful to have a cmdline switch to turn off stdin reading altogether..

**Comment by firnsan** [comment]
I think the only way is to write a expect script to wrap the mcserver.
I doubt that your team have ever run mcserver for long time, and play game on it. Because morest of my servers would stop login session for kinds of reason, nohup is a best way to run program for a long time.But mcserver dont support this way...

**Comment by sphinxc0re** [comment]
You can start the server as a daemon:
```bash
./Cuberite -d
```
It won't accept any input from there on

**Comment by firnsan** [comment]
Yeah，that is great , thank you \^O^/

**Comment by LogicParrot** [comment]
You could always use `tmux` or `screen`, too.

As for the original issue: Actually, running away when hit is implemented. Chickens should run away when hit. However, when holding food, chickens priorotize the food and do not escape. #3295 should fix that part.

**Comment by mathiascode** [comment]
Duplicate of #2554
